### ISSUE TITLE
Option to change user details - name, email, password

### ISSUE THREAD

**Issue by jhenshall** [mapswipe/mapswipe#251]
Can we add the ability for users to update their details in-app including their username, email address and password?

We've had a couple of requests for this via email and currently have to reply that it's not possible to change email or username and the process for password requires signing out and reseting.

[Seems possible from Firebase](https://firebase.google.com/docs/reference/js/firebase.User#updateemail). Need to confirm no GDPR considerations.

**Comment by laurentS** [comment]
Good point. I think this is probably going in the general direction of GDPR anyway, as I think there's a requirement to allow users to modify/delete their personal data.
@Hagellach37 if users modify their username, is there any impact to consider on data beyond their user profile? I can't think of any other place where we store the username.

**Comment by Hagellach37** [comment]
From the data side the most important information is the id which should not change. 
We have the username also in the postgres db. We had used it once to display the username for the MapSwipeLive website, but since this is not up-to-date anymore, we currently just store it in postgres and do nothing further. We can even think about deleting it there if we do not have any use case in mind.

**Comment by laurentS** [comment]
So if we allow a user to change their username, we shouldn't be breaking anything at this point, if I understand correctly.

**Comment by Hagellach37** [comment]
yep, this will not break anything. To begin with we can also ignore the username not being updated in postgres, no problem. So this could be handled from the app side entirely.

**Comment by laurentS** [comment]
Pushing this up a little in priority as we've seen a number of requests for this (particularly change of email) through the info@ email.
